# This assignment is an individual work
- Author: Junru Lu(lj1230)

### IDEA: The implementation of an alternative bus route for bus line X8 improves circulation
### NULL HYPOTHESIS: The commute time is _the same_ or _longer_ with the new bus route as it was before
### ALTERNATIVE HYPOTHESIS: The commute time is _shorter_ with the new bus route as it was before
#### _$H_0$_ : $T_{\mathrm{old}} - T_{\mathrm{new}} <= 0 $
#### _$H_1$_ : $T_{\mathrm{old}} - T_{\mathrm{new}} > 0 $
### Significance level: $\alpha=0.05$
### I'm going to use Z-test: $z = \frac{\mu_{\mathrm{pop}}-\mu_{\mathrm{sample}}}{\sigma/\sqrt{N}} $ for this hypnosis testing

In [1]:
# download data
!curl -O https://raw.githubusercontent.com/fedhere/PUI2018_fb55/master/Lab4_fb55/times.txt

  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100  1179  100  1179    0     0   4122      0 --:--:-- --:--:-- --:--:--  4136


In [2]:
import os
import pandas as pd

# test if $PUIDATA is there. If not, create it
if os.getenv("PUIDATA") is None:
    os.environ["PUIDATA"] = "%s/PUIdata"%os.getenv("HOME")
else:
    pass
print(os.getenv("PUIDATA"))

/nfshome/lj1230/PUIdata


In [3]:
# move data to PUIdata folder
!mv times.txt $PUIDATA

In [4]:
# statistical description of old bus time
old_mu = 36
old_std = 6

# load data
df = pd.read_table(os.getenv("PUIDATA") + "/times.txt", 'r', names=['times'])

In [5]:
df.head(5)

times
0  31.622239
1  32.821376
2  30.229101
3  31.413766
4  39.010550

In [6]:
df.shape

(100, 1)

In [7]:
mu = df['times'].mean()
std = df['times'].std()
mu, std

(34.466161688299998, 7.1372801337518794)

In [8]:
size = df.shape
size[0]

100

In [9]:
# Z statistics is the number of standard deviations away from the mean of your sample
z = (old_mu-mu)/(old_std/size[0] ** 0.5)
z

2.5563971861666701

### So the conclusion that z score = 2.56 > 2 means the new mean are further than $2\sigma$ away from the old mean, based on which we can reject the Null Hypothesis above at the 95% confidence level